MKL_README.md

```diff
@@ -1,3 +1,21 @@
+# Full MKL Installation
```

Comment:
You should mention the purpose of doing so, i.e. that this will enable MKL for all operators in the linalg-namespace.

What about this piece of code (I guess it is still in the config):
# by default, disable lapack when using MKL
# switch on when there is a full installation of MKL available (not just MKL2017/MKL_ML)
ifeq ($(USE_BLAS), mkl)
USE_LAPACK = 0
endif

Guess this has to be changed as well. 

And unfortunately this does not work exactly as planned. With the suggested settings, a user would get MKL for blas/lapack, but same time setting USE_MKL2017=0 would internally switch off use of MKLML for a alot of NN operators.  Setting USE_MKL2017=0 was just a shortcut for our experiments with the linalg-operators. 

The mechanism ideally should work such that the user just sets USE_BLAS=mkl and that is it. He can in addition set USE_MKL2017 and then also some other operators will start useing MKL's NN-functions.

Reply:
Okay, I believe I changed the config file to adhere to the behavior we want.

Reply:
There seems to me no config changes now except removal of the piece of code above. Is this enough? I.e. have you tried that specifically the variant
USE_BLAS=mkl
USE_MKL2017=1
works?